src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,342 @@
+/*
+ * This file is part of the Soletta™ Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "sol-flow/netctl.h"
+#include "sol-flow-internal.h"
+
+#include <sol-netctl.h>
+#include <sol-str-table.h>
+#include <sol-util-internal.h>
+#include <errno.h>
+
+struct list_service_data {
+    struct sol_ptr_vector service_list;
```

Comment:
`list_service_data` and its `service_list` shouldn't be per node, since all nodes will have the exact same content. Actually you do not need to maintain a shadow/copy list, since it will be the same as `sol_netctl_get_services()` (https://github.com/solettaproject/soletta/blob/master/src/lib/comms/include/sol-netctl.h#L472).

Then from `network_service_cb` you simply emit flow packets for that node (you give it as `void *data`).